r-package/grf/R/causal_forest.R

```diff
@@ -197,88 +202,56 @@ causal_forest <- function(X, Y, W,
-      mtry = mtry,
-      alpha = alpha,
-      imbalance.penalty = imbalance.penalty,
-      stabilize.splits = stabilize.splits,
-      num.threads = num.threads,
-      honesty = honesty,
-      honesty.fraction = honesty.fraction,
-      honesty.prune.leaves = honesty.prune.leaves,
-      seed = seed,
-      clusters = clusters,
-      samples.per.cluster = samples.per.cluster
-    )
-    tunable.params <- tuning.output$params
-  } else {
-    tunable.params <- c(
-      min.node.size = validate_min_node_size(min.node.size),
-      sample.fraction = validate_sample_fraction(sample.fraction),
-      mtry = validate_mtry(mtry, X),
-      alpha = validate_alpha(alpha),
-      imbalance.penalty = validate_imbalance_penalty(imbalance.penalty),
-      honesty.fraction = validate_honesty_fraction(honesty.fraction, honesty),
-      honesty.prune.leaves = validate_honesty_prune_leaves(honesty.prune.leaves)
-    )
-  }
-
   Y.centered <- Y - Y.hat
   W.centered <- W - W.hat
+  data <- create_data_matrices(X, outcome = Y.centered, treatment = W.centered,
+                              sample.weights = sample.weights)
+  args <- list(clusters = clusters,
+              samples.per.cluster = samples.per.cluster,
+              sample.fraction = sample.fraction,
+              mtry = mtry,
+              min.node.size = min.node.size,
+              honesty = honesty,
+              honesty.fraction = honesty.fraction,
+              honesty.prune.leaves = honesty.prune.leaves,
+              alpha = alpha,
+              imbalance.penalty = imbalance.penalty,
+              stabilize.splits = stabilize.splits,
+              ci.group.size = ci.group.size,
+              num.threads = num.threads,
+              seed = seed,
+              reduced.form.weight = 0)
 
-  data <- create_data_matrices(X, outcome = Y.centered, treatment = W.centered, sample.weights = sample.weights)
-  forest <- causal_train(
-    data$train.matrix, data$sparse.train.matrix,
-    data$outcome.index, data$treatment.index, data$sample.weight.index,
-    data$use.sample.weights,
-    as.numeric(tunable.params["mtry"]),
-    num.trees,
-    as.numeric(tunable.params["min.node.size"]),
-    as.numeric(tunable.params["sample.fraction"]),
-    honesty,
-    as.numeric(tunable.params["honesty.fraction"]),
-    as.numeric(tunable.params["honesty.prune.leaves"]),
-    ci.group.size,
-    reduced.form.weight,
-    as.numeric(tunable.params["alpha"]),
-    as.numeric(tunable.params["imbalance.penalty"]),
-    stabilize.splits,
-    clusters,
-    samples.per.cluster,
-    compute.oob.predictions,
-    num.threads,
-    seed
-  )
+  if (tune.parameters[1L] != "none") {
+    if (tune.parameters[1L] == "all") {
+      tune.parameters <- tunable.params
+      if (!honesty) {
+        tune.parameters <- tune.parameters[!grepl("honesty", tune.parameters)]
+      }
+    } else {
+      tune.parameters <- unique(match.arg(tune.parameters, tunable.params, several.ok = TRUE))
```

Comment:
At first glance, I was momentarily confused by the meaning of `tunable.params` vs `tune.parameters`.

I'm not sure what's the best option here. How about at least changing `tunable.params` to `all.tunable.params`?